include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h

```diff
@@ -171,7 +171,32 @@ struct PaddingToAvoidFalseSharing {
 4. Note LogStart must pair with either LogEnd or LogEndAndStart, otherwise ORT_ENFORCE will fail;
 5. ThreadPoolProfiler is thread-safe.
 */
-
+#ifdef ORT_MINIMAL_BUILD
+class ThreadPoolProfiler {
+ public:
```

Comment:
What's the reason why ThreadPoolProfiler doesn't have it's own .cc and .h file?

Reply:
We believe it is to be consumed solely by the thread pool instead of any other entity. 
Even when the executor trying to get the profiled numbers, it is done by calling thread pool interface rather than anyone from the profiler.

Reply:
Would it make sense to put it in include/onnxruntime/core/platform/detail/thread_pool_profiler.h so that's clear (could also be in onnxruntime.detail namespace) and we keep it to one significant class per header/file? 

At the very least can we put the implementation in thread_pool_profiler.cc?



---
In reply to: [600065799](https://github.com/microsoft/onnxruntime/pull/7115#discussion_r600065799) [](ancestors = 600065799)

Reply:
thread_pool_profiler.h looks all good - we are refactoring the entire EigenNon...h header to move out implementation details to separate cc files. Will go with "thread_pool_profiler.h" approach in a PR underway.